rspace/src/main/scala/coop/rchain/rspace/history/KeySegmentBase.scala

```diff
@@ -0,0 +1,23 @@
+package coop.rchain.rspace.history
+import scodec.bits.ByteVector
+import scala.annotation.tailrec
+
+trait KeySegmentBase[A] {
+  val value: A
+  def size: Long
+  def isEmpty: Boolean
+  def nonEmpty: Boolean
+  def head: Byte
```

Comment:
What are other possible implementations for KeySegmentBase then ByteVector?
Here you are making a type parameter A, but assume that head is a `Byte`, and tail is `KeySegmentBase[A]`. Does not this mean that all implementations of `KeySegmentBase[A]` are sequences of bytes?